### PR TITLE
use standard token from open zeppelin and set up truffle

### PR DESCRIPTION
**Pull request by cpopov**
- set up truffle
- set up linting for ES6
- Standard Token

### CODE REVIEW

**Comment by microdesign** [comment]
Great

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+root = true
+
+[*]
+charset = utf-8
+indent_style = space
+indent_size = 2
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
```
Comment (raddrick) [inline .editorconfig]: this is a personal config, it doesnt need to be commited to the repo as it is only for your config

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    // Strict mode
+    "strict": [
+      2,
+      "global"
+    ],
+
+    // Code style
+    "space-before-function-paren": "off"
+  }
+}
```
Comment (raddrick) [inline .eslintrc]: same with this file. i have my

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+*.ipr
+
+#Mac
+.DS_Store
+
+#Truffle
+build/
+secrets.json
+coverage.json
+coverage/
```
Comment (raddrick) [inline .gitignore]: here as well, this  should be in your global gitignore

File: .solhint.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
+  "extends": "default",
+  "rules": {
+    "avoid-throw": false,
+    "avoid-suicide": "error",
+    "avoid-sha3": "warn",
+    "indent": ["warn", 2]
+  }
+}
```
Comment (raddrick) [inline .solhint.json]: samesies

File: contracts/Erc20.sol [hunk 5]
```diff
@@ -1,13 +0,0 @@
-pragma solidity ^0.4.24;
-
-contract Erc20 {
-  uint public _totalSupply;
```
Comment (raddrick) [inline contracts/Erc20.sol]: why not just update the TotalSupply interface instead of copy and pasting an entirely new file?

File: contracts/ERC20.sol [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    address indexed to,
+    uint256 value
+  );
+
+  event Approval(
+    address indexed owner,
+    address indexed spender,
+    uint256 value
+  );
+}
```
Comment (raddrick) [inline contracts/ERC20.sol]: i would prefer the formatting be consistent, line 9 here could be in the Erc20.sol and we would have a consistant codebase and a leaner commit

File: contracts/EvedoToken.sol [hunk 7]
```diff
@@ -1,122 +1,61 @@
...
+    totalSupply_ = TOTAL_SUPPLY;
   }
 
-  /**
-   * Just being sent some cash? Let's buy tokens
-   */
-  function() public payable {
-    buy();
-  }
 }
```
Comment (raddrick) [inline contracts/EvedoToken.sol]: this is hard to parse this way to review..i assume its all backed up by the tests though, so its okay

File: contracts/EvedoTokenSale.sol [hunk 8]
```diff
@@ -292,6 +297,16 @@ contract EvedoTokenSale {
...
+  modifier isAvailable() {
+    require(!halted && !freeze);
+    _;
+  }
+
+  modifier onlyCompany() {
+    require(msg.sender == company);
+    _;
+  }
+
```
Comment (raddrick) [inline contracts/EvedoTokenSale.sol]: pretty sure these were defined already...maybe moved?

File: .editorconfig (hunk 1 above)
Comment (cpopov) [reply .editorconfig]: I don't think so. This is to ensure we all the same formatting. This is supposed to be IDE agnostic. Do you know of any alternatives?

File: .eslintrc (hunk 2 above)
Comment (cpopov) [reply .eslintrc]: No, we need to all have the same esling conf. See a good example of a solidity project here https://github.com/gnosis/pm-contracts

File: .solhint.json (hunk 4 above)
Comment (cpopov) [reply .solhint.json]: This is for solhint - I think very good practice to have it

File: contracts/ERC20.sol (hunk 6 above)
Comment (cpopov) [reply contracts/ERC20.sol]: Sorry, This was the formatting from openzepelin, not my choice

File: .gitignore (hunk 3 above)
Comment (cpopov) [reply .gitignore]: I think we should have this in the repo. Every single project I have worked for uses .gitignore. This is shared between all of use.  see https://github.com/gnosis/pm-contracts/blob/development/.gitignore
